### PR TITLE
Add ScrollToTop component

### LINKED ISSUE #1200: Bug - Navigating to new page should jump to the top of the page
# Description
- When navigating from the Search page to the Request page, after hitting "Request Candidates" the form. The page does not jump to the top of the page. I believe we have a [component available on TalentCloud](https://github.com/GCTC-NTGC/TalentCloud/blob/dev/resources/assets/js/components/ScrollToTop.tsx) that can solve this.

### PR DESCRIPTION
**Pull request by mnigh**
Resolves #1200.

- Adds ScrollToTop component to **common**
- Adds ScrollToTop component to **talentsearch**

### CODE REVIEW

File: common/src/helpers/router.tsx [hunk 1]
```diff
@@ -157,3 +157,41 @@ export const Link: React.FC<{ href: string; title: string }> = ({
...
+interface ScrollToTopProps {
+  /** The top position of element relative to the viewport.  */
+  offsetTop?: number;
+  /** Set the scroll behavior to 'auto'. Default = 'smooth' */
+  scrollBehaviorAuto?: boolean;
+  children: React.ReactElement | null;
+}
+
+export const ScrollToTop: React.FC<ScrollToTopProps> = ({
+  offsetTop,
```
Comment (brindasasi) [inline common/src/helpers/router.tsx:170]: minor tip! if you define default value for offsetTop here `offsetTop=0` , you could eliminate the `if else`  below

Comment (mnigh) [reply common/src/helpers/router.tsx]: > minor tip! if you define default value for offsetTop here offsetTop=0 , you could eliminate the if else below

Thanks @brindasasi! I was trying to figure out a way to simplify or get rid of the conditional statements  and this solution is succinct.

**Review by yonikid15 — COMMENTED** [review]
I think we can simplify this component a lot more now that I've looked at it. Most of the extra stuff we had in there was overkill needed for TalentCloud. This component should have no props, except for `children`, which you can remove the `null` type and just set the return type to `return children | null`. All the inner part of the component needs should be this useEffect hook:

```js
  useEffect(() => {
    window.scrollTo({
      top: 0,
      left: 0,
      behavior: "auto",
    });
  });
```

**Comment by mnigh** [comment]
> I think we can simplify this component a lot more now that I've looked at it. Most of the extra stuff we had in there was overkill needed for TalentCloud. This component should have no props, except for `children`, which you can remove the `null` type and just set the return type to `return children | null`. All the inner part of the component needs should be this useEffect hook:
> 
> ```js
>   useEffect(() => {
>     window.scrollTo({
>       top: 0,
>       left: 0,
>       behavior: "auto",
>     });
>   });
> ```

Indeed, makes sense if we the top is really the top of the document and not some element. So you think the scrolling should be `behavior: "auto"` and not `behavior: "smooth"` like Talent Cloud?

**Review by yonikid15 — APPROVED** [review]
Looks good! 👍

**Pull request merged**